diff --git a/ulauncher/api/shared/action/LaunchAppAction.py b/ulauncher/api/shared/action/LaunchAppAction.py
index 95ce7d8c..e3a28103 100644
--- a/ulauncher/api/shared/action/LaunchAppAction.py
+++ b/ulauncher/api/shared/action/LaunchAppAction.py
@@ -42 +42 @@ class LaunchAppAction(BaseAction):
-            exec = ['gapplication', 'launch', app_id]
+            exec = ['uwsm', 'app', '--', app_id]
@@ -52 +52 @@ class LaunchAppAction(BaseAction):
-                    exec = ['gtk-launch', app_id]
+                    exec = ['uwsm', 'app', '--', app_id]
@@ -54 +54 @@ class LaunchAppAction(BaseAction):
-                exec = shlex.split(stripped_app_exec)
+                exec = ['uwsm', 'app', '--' ] + shlex.split(stripped_app_exec)
@@ -64 +64 @@ class LaunchAppAction(BaseAction):
-                exec = ['systemd-run', '--user', '--scope'] + exec
+                exec = ['uwsm', 'app', '--'] + exec
