diff --git a/debug/gdb/printers.py b/debug/gdb/printers.py
index 24961d115..9de882e33 100644
--- a/debug/gdb/printers.py
+++ b/debug/gdb/printers.py
@@ -83,7 +83,7 @@ class EigenMatrixPrinter:
 			type = type.target()
 		self.type = type.unqualified().strip_typedefs()
 		tag = self.type.tag
-		regex = re.compile('\<.*\>')
+		regex = re.compile('<.*>')
 		m = regex.findall(tag)[0][1:-1]
 		template_params = m.split(',')
 		template_params = [x.replace(" ", "") for x in template_params]
@@ -150,7 +150,7 @@ class EigenSparseMatrixPrinter:
 			type = type.target()
 		self.type = type.unqualified().strip_typedefs()
 		tag = self.type.tag
-		regex = re.compile('\<.*\>')
+		regex = re.compile('<.*>')
 		m = regex.findall(tag)[0][1:-1]
 		template_params = m.split(',')
 		template_params = [x.replace(" ", "") for x in template_params]
